packages/runtime/src/ctx.ts

```diff
@@ -215,8 +215,9 @@ export class Ctx implements Context {
 
 	getCallerApplicationID(): number {
 		let callerApplicationID = 0;
-		if (this.innerTxAppIDCallStack.length > 0) {
-			callerApplicationID = this.innerTxAppIDCallStack[this.innerTxAppIDCallStack.length - 1];
+		// the top value in the `innerTxAppIDCallStack` is the own appID not the caller appID
+		if (this.innerTxAppIDCallStack.length > 1) {
```

Comment:
It seems you are using my code from https://github.com/scale-it/algo-builder/pull/798/files but not with my commit nor adding me as a co-author. This is not very fair.